examples/nidaqmx/analog-input-every-n-samples-aio.py

```diff
@@ -101,15 +101,21 @@ def check_for_warning(response):
                 )
             )
 
-            # Wait for initial_metadata to ensure that the callback is registered before starting
-            # the task.
+            # Wait for initial_metadata and check for stream errors to ensure that the callback is
+            # registered before starting the task.
             await every_n_samples_stream.initial_metadata()
+            if every_n_samples_stream.done():
+                every_n_samples_response = await every_n_samples_stream.read()
```

Comment:
I should test what happens if the event stream is closed with an OK status. 

For aio, I think `read()` returns a [`grpc.aio.EOF`](https://github.com/grpc/grpc/blob/cf6e5ed8434a260ccc8d526cc86a016b34afee49/src/python/grpcio/grpc/_cython/_cygrpc/aio/common.pyx.pxi#L57), which seems like some kind of placeholder.

For non-aio, I'm using `next()`, which probably raises `StopIteration`. Specifying a default value should prevent that.

Reply:
aio calls don't have an `exception()` method, so I used `await stream.code()` to check for errors. I still have to use `await stream.read()` to raise an exception, so I extracted a `check_for_stream_error()` helper function.